zebra/zebra_nhg.c

```diff
@@ -1959,6 +1960,52 @@ static int resolve_backup_nexthops(const struct nexthop *nexthop,
 	return 0;
 }
 
+/*
+ * This code is depending on the fact that a nexthop->ifindex is 0
+ * if it is not known, if this assumption changes, yummy!
+ */
+static struct nexthop *zebra_nhg_connected_ifindex(struct route_node *rn,
+						   struct route_entry *match,
+						   int32_t curr_ifindex)
+{
+	struct nexthop *newhop = match->nhe->nhg.nexthop;
+	struct route_entry *re;
+
+	assert(newhop); /* What a kick in the patooey */
+
+	if (curr_ifindex == 0)
+		return newhop;
+
+	if (curr_ifindex == newhop->ifindex)
+		return newhop;
+
+	/*
+	 * At this point we know that this route is matching a connected
+	 * but there are possibly a bunch of connected routes that are
+	 * alive that should be considered as well.  So let's iterate over
+	 * all the re's and see if they are connected as well and maybe one
+	 * of those ifindexes match as well.
+	 */
+	RNODE_FOREACH_RE (rn, re) {
+		if (re->type != ZEBRA_ROUTE_CONNECT)
+			continue;
+
+		if (CHECK_FLAG(re->status, ROUTE_ENTRY_REMOVED))
```

Comment:
should we confirm INSTALLED status here also?

Reply:
can you point me to a connected route that isn't installed by it's very nature?